packages/twenty-front/src/modules/ui/navigation/navigation-drawer/components/NavigationDrawerCollapseButton.tsx

```diff
@@ -28,9 +30,22 @@ export const NavigationDrawerCollapseButton = ({
   className,
   direction = 'left',
 }: NavigationDrawerCollapseButtonProps) => {
+  const isSettingsPage = useIsSettingsPage();
+
   const setIsNavigationDrawerExpanded = useSetRecoilState(
     isNavigationDrawerExpandedState,
   );
+  const isNavigationDrawerExpanded = useRecoilValue(
+    isNavigationDrawerExpandedState,
+  );
+
+  // Check if the user is on the settings page and the navigation drawer is not expanded
+  // Then open it automatically
+  useEffect(() => {
+    if (!isNavigationDrawerExpanded && isSettingsPage) {
```

Comment:
we don't put side effects in components as we don't want to encourage developers to implement behaviors during rendering (this is pushing them to take dependencies on variables and trigger re-renders). Instead, we either:
- implement sync behaviors (in callbacks, event handlers, ...)
- move useEffects to ComponentEffects that have no children so it's not a big deal from a performance perspective

Reply:
Right. I should be creating a custom hook to achieve it, right?